benchmarks/parser.py

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+"""Benchmarks for the parser of the xDSL implementation."""
+
+from benchmarks.workloads import WorkloadBuilder
+from xdsl.context import Context
+from xdsl.dialects.arith import Arith
+from xdsl.dialects.builtin import Builtin
+from xdsl.parser import Parser as XdslParser
+
+CTX = Context(allow_unregistered=True)
+CTX.load_dialect(Arith)
+CTX.load_dialect(Builtin)
+
+
+class Parser:
+    """Benchmark the xDSL parser on MLIR files."""
+
+    WORKLOAD_CONSTANT_100 = WorkloadBuilder.constant_folding(100)
+    WORKLOAD_CONSTANT_1000 = WorkloadBuilder.constant_folding(1000)
+    WORKLOAD_LARGE_DENSE_ATTR = WorkloadBuilder.large_dense_attr()
+    WORKLOAD_LARGE_DENSE_ATTR_HEX = WorkloadBuilder.large_dense_attr_hex()
+
+    def time_constant_100(self) -> None:
+        """Time lexing constant folding for 100 items."""
+        XdslParser(CTX, Parser.WORKLOAD_CONSTANT_100).parse_module()
+
+    def time_constant_1000(self) -> None:
+        """Time lexing constant folding for 1000 items."""
+        XdslParser(CTX, Parser.WORKLOAD_CONSTANT_1000).parse_module()
+
+    def ignore_time_dense_attr(self) -> None:
+        """Time lexing a 1024x1024xi8 dense attribute."""
+        XdslParser(CTX, Parser.WORKLOAD_LARGE_DENSE_ATTR).parse_module()
+
+    def ignore_time_dense_attr_hex(self) -> None:
+        """Time lexing a 1024x1024xi8 dense attribute given as a hex string."""
+        XdslParser(CTX, Parser.WORKLOAD_LARGE_DENSE_ATTR_HEX).parse_module()
+
+
+if __name__ == "__main__":
+    from bench_utils import Benchmark, profile
```

Comment:
Quick question for my knowledge, why is this import here and not at the toplevel?